ofrak_core/ofrak/model/resource_model.py

```diff
@@ -894,3 +894,22 @@ def _validate_indexed_type(getter_func: Callable[[Any], X]):
             f"Type of index {getter_func.__name__} is {index_type}, which is not "
             f"one of {_INDEXABLE_TYPES.values()}; cannot index by this value!"
         )
+
+
+@dataclasses.dataclass(**ResourceAttributes.DATACLASS_PARAMS)
+class DataBytes(ResourceAttributes):
+    """
+    Special attributes class for accessing info about a resource's binary data.
+    SHOULD NEVER BE USED OUTSIDE OF INTERNAL OFRAK MECHANISMS.
+    """
+
+    _offset: int
+    _length: int
```

Comment:
These attributes are going to be useful in the GUI. With that in in mind, I think there are three things I would like to know:

1. Is the data mapped?
2. What is the offset in the root?
3. What is the size of the data?

I'm wondering if this attribute could contain all three pieces of information. Something like:

```
@dataclass
class DataBytes

    offset_in_root: Optional[int]
    length: int
```

What do you think?

Reply:
It's an interesting idea. What are your thoughts on how extensively we should rely on/use these sort-of-magic attributes? I think the more visible we make this class, the more protections we need against manual (or accidental) modification of these attributes.

Reply:
FWIW this will show up in the GUI.

I think we make the dataclass frozen and update the docstring to say explicitly that setting these fields will not update the underlying data can go a long way.

Reply:
The dataclass was already frozen, and I elaborated in the docstring as you suggested.

Per our quick offline chat about this, showing this in the GUI is helpful since it's the only way to see data offset and length, so I won't add code to filter it out.